File: September 2026/2026-09-08/43-multiply-string.py
# 43-multiply-string.cpp;maths-string;help;leetcode

class Solution:
    def multiply(self, num1: str, num2: str) -> str:
        if num1 == "0" or num2 == "0":
            return "0"
        m = len(num1)
        n = len(num2)

        result = [0] * (m + n)

        for i in range(m - 1, -1, -1):
            for j in range(n - 1, -1, -1):
                a = ord(num1[i]) - ord('0')
                b = ord(num2[i]) - ord('0')

                product = a * b
                pos1 = i + j
                pos2 = i + j + 1
                total = product + result[pos2]

                result[pos2] = total % 10
                result[pos1] += total // 10

        i = 0
        while i < len(result) and result[i] == 0:
            i += 1

        ans = ""

        while i < len(result):
            ans += chr(result[i] + ord('0'))
            i += 1

        return ans

File: September 2026/2026-09-08/43-multiply-string.cpp
// 43-multiply-string.cpp;maths-string;help;leetcode

class Solution {
public:
    string multiply(string num1, string num2) {
        if (num1 == "0" || num2 == "0")
            return "0";
        int m = num1.size();
        int n = num2.size();

        vector<int> result(m + n, 0);

        for(int i = m - 1; i >= 0; i--) {
            for(int j = n - 1; j >= 0; j--) {
                int a = num1[i] - '0';
                int b = num2[j] - '0';

                int product = a * b;
                int pos1 = i + j;
                int pos2 = i + j + 1;

                int sum = product + result[pos2];

                result[pos2] = sum % 10;
                result[pos1] += sum / 10;
            }
        }

        string ans = "";
        int i = 0;

        while(i < result.size() && result[i] == 0) {
            i++;
        }

        while(i < result.size()) {
            ans += char(result[i] + '0');
            i++;
        }

        return ans;
    }
};